packages/api-core/src/resources/filesDelivery.js

```diff
@@ -0,0 +1,39 @@
+import AvMicroservice from '../ms';
+
+export default class AvFilesDelivery extends AvMicroservice {
+  constructor({ http, promise, merge, config }) {
+    const options = Object.assign(
+      {
+        name: 'platform/file-upload-delivery/v1/batch/deliveries',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        polling: true,
+        pollingMethod: 'GET',
+      },
+      config
+    );
+    super({
+      http,
+      promise,
+      merge,
+      config: options,
+    });
+  }
+
+  uploadFilesDelivery(data, config) {
+    if (!config.customerId || !config.clientId) {
```

Comment:
You should do this check after you merge the configs via `config = this.config(config);`.
If the user wants to provide the `clientId` (or even `customerId`) in the initial config provided to the constructor and not with every call to `uploadFilesDelivery`, it should be allowed.